### ISSUE TITLE
Komodo IDE 9: Issues with Perl Critic background syntax checking

### ISSUE THREAD

**Issue by fjossandon** [Komodo/KomodoEdit#87]
Hello,
I'm using the Perl Critic option integrated in Komodo (in Windows 8.1, x64) for the background syntax checking to produce additional warnings, and I use the ".perlcriticrc" to change some defaults parameters, but there are 2 problems with it.

Perl Critic by default checks if the file is a module (to check if it must apply module-specific policies) by first looking for the presence of the Shebang in the very first line (#!/usr/bin/perl), and if that fails it looks if the file extension is ".PL" (case sensitive)  for the sake of special files like Makefile.PL which are always scripts (not modules) but usually don't have a shebang line; so if the file doesn't have a shebang and its not one of the special .PL files, then Perl::Critic treats ii like a module (as the author explained me).

I have configured Critic on severity level 4 ('Stern') and the first problem is that I see 2 module warnings (`Code not contained in explicit package` and `Module does not end with "1;"`), that indicates me that Critic thinks the script is a module even if it have the ".PL" extension when the shebang is ABSENT. This only happens in the warnings generated by Komodo, and DOES NOT happen if I launch Critic manually using code:

``` Perl
use Perl::Critic;
my $file = 'script.PL';
my $critic = Perl::Critic->new(-severity => 4);
my @violations = $critic->critique($file);
print @violations; # No module warnings should be produced
exit;
```

So something in Komodo is making Critic ignore the ".PL" extension, and only the shebang check is performed correctly.

The second problem (which probably relates to the first one) is that the ".perlcriticrc" file can be configured to add case sensitive extensions that will be taken by Critic as program files instead of modules (besides the default ".PL"). Writing the `program-extensions = .pl` parameter in the file have no effect in making the ".pl" extension to not be treated like a module (again when the script don't use the shebang for the purpose of checking this), and Komodo still prints the module warnings mentioned above. This does not happen when using Perl Critic through code, where the warnings correctly dissappear:

``` Perl
use Perl::Critic;
my $file = 'script.pl';
my $critic = Perl::Critic->new(-severity => 4,
                               -profile => '.perlcriticrc');
my @violations = $critic->critique($file);
print @violations; # No module warnings should be produced
exit;
```

In summary, when running separately, Perl Critic is able to detect the script as a program and not a module using the file extensions when the shebang is missing, but it fails when running in background automatically through Komodo. I have no clue why this is happening.

Best regards.

**Comment by fjossandon** [comment]
Closing my issue since the repo is no longer maintained

**Comment by th3coop** [comment]
Thanks for that @fjossandon.  You don't have to obviously but thank you regardless.

**Comment by fjossandon** [comment]
Hey, no worries. Today I was looking at my open issues and decided to close those that were too old, and 9 years is too much time to keep a ticket open. =)
